it/controllers/BusinessNameControllerISpec.scala

```diff
@@ -0,0 +1,89 @@
+
+package controllers
+
+import connectors.stubs.IncomeTaxSubscriptionConnectorStub._
+import helpers.ComponentSpecBase
+import helpers.IntegrationTestConstants._
+import helpers.servicemocks.AuthStub.stubAuthSuccess
+import play.api.http.Status._
+import play.api.libs.json.Json
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.BusinessNameController
+
+class BusinessNameControllerISpec extends ComponentSpecBase {
+
+  "GET /report-quarterly/income-and-expenses/sign-up/self-employments/details/business-name" when {
+
+    "the Connector is empty" should {
+      "return the page with no prepopulated fields" in {
+        Given("I setup the Wiremock stubs")
+        stubAuthSuccess()
+        stubGetSelfEmployments(BusinessNameController.businessNameKey)(NO_CONTENT)
+
+        When("GET /details/business-name is called")
+        val res = getBusinessName()
+
+        Then("should return an OK with the BusinessNamePage")
+        res must have(
+          httpStatus(OK),
+          pageTitle("What is the name of your business?")
+        )
+      }
+    }
+    "Connector returns a previously filled in Business Name" should {
+      "show the current business name page with the name value entered" in {
+        Given("I setup the Wiremock stubs")
+        stubAuthSuccess()
+        stubGetSelfEmployments(BusinessNameController.businessNameKey)(OK, Json.toJson(testBusinessNameModel))
+
+        When("GET /details/business-name is called")
+        val res = getBusinessName()
+
+        Then("should return an OK with the BusinessNamePage")
+        res must have(
+          httpStatus(OK),
+          pageTitle("What is the name of your business?"),
+          textField("businessName", testBusinessName)
+
+        )
+
+
+      }
+
+
+    }
+
+
+  }
+  "POST /report-quarterly/income-and-expenses/sign-up/self-employments/details/business-name" when {
+    "the form data is valid and is stored successfully" in {
+      Given("I setup the Wiremock stubs")
+      stubAuthSuccess()
+      stubSaveSelfEmployments(BusinessNameController.businessNameKey)(CREATED)
```

Comment:
There is a second parameter to the stubSaveSelfEmployments function which is the data you are saving, could you provide this parameter